salt/metalk8s/orchestrate/deploy_new_node.sls

```diff
@@ -1,47 +1,53 @@
-{%- if pillar['bootstrap_id'] %}
-{%-   set control_plane_ips = salt.saltutil.runner('mine.get', tgt=pillar['bootstrap_id'], fun='control_plane_ip') %}
-{%- else %}
-{%-   set control_plane_ips = {} %}
-{%- endif %}
+{%- set control_plane_ip = salt.saltutil.runner(
+    'mine.get',
+    tgt=pillar['bootstrap_id'],
+    fun='control_plane_ip'
+)[pillar['bootstrap_id']] %}
 
-{%- if pillar['bootstrap_id'] in control_plane_ips.keys() and control_plane_ips[pillar['bootstrap_id']] %}
-{%-   set control_plane_ip = control_plane_ips[pillar['bootstrap_id']] %}
-{%- else %}
-{%-   set control_plane_ip = 'localhost' %}
-{%- endif %}
+{%- set pillar_data = {
+    'repo': {
+        'host': control_plane_ip
+    },
+    'registry_ip': control_plane_ip
+} %}
 
+{%- set version = pillar.metalk8s.nodes[pillar.bootstrap_id].version %}
 
+{%- if pillar['node_name'] not in salt['saltutil.runner']('manage.up') %}
 Deploy salt-minion on a new node:
   salt.state:
     - ssh: true
     - roster: kubernetes
     - tgt: {{ pillar['node_name'] }}
-    - saltenv: {{ saltenv }}
+    - saltenv: metalk8s-{{ version }}
     - sls:
       - metalk8s.roles.minion
-    - pillar:
-        repo:
-          host: {{ control_plane_ip }}
-        salt:
-          master:
-            host: {{ control_plane_ip }}
+    - pillar: {{ pillar_data | tojson }}
 
 Accept key:
-  salt.state:
-    - tgt: {{ pillar['bootstrap_id'] }}
-    - saltenv: {{ saltenv }}
-    - sls:
-      - metalk8s.salt.master.accept_keys
+  module.run:
```

Comment:
Strictly speaking I believe you first have to wait for the node to attempt to authenticate (`wait_for_event`).

Reply:
Yes I know but as we will likely not use this in the futur lets not do this for the moment.
In the minion salt states we wait for the salt-minion to be really started so should be ok